### DISCUSSION TITLE
Extend HTML Report to say how many asserts were used in each test case

### DISCUSSION THREAD

**Discussion by nagkumar** [junit-team/junit-framework#3300] [Q&A]
![image](https://github.com/junit-team/junit5/assets/332234/7b43f391-ff47-4fd4-bff2-683fd582d2fe)

In the above report I want to add total asserts used in that test case.

I was able to count asserts using AOP way and out come is such as

  ```
com.tejasoft.tests.ju.ju5.ut.suites.learn.TestSuiteEx

    JUnit Jupiter

      TestExAdd

        ΓêÜ testAddition()

        Group: JU3
        Total 'assertEquals' Count: 1
        Total 'assertFalse' Count: 1
        Total 'assert*' Count: 2

        Group: JU4
        Total 'assertEquals' Count: 102
        Total 'assert*' Count: 103

        Group: JU5 Asserts
        Total 'assertEquals' Count: 773
        Total 'assert*' Count: 775

        Group: JU5 Assumes

        Group: TNG
        Total 'assertEquals' Count: 1
        Total 'assertTrue' Count: 1
        Total 'assertFalse' Count: 1
        Total 'assertNotNull' Count: 1
        Total 'assertNull' Count: 1
        Total 'assert*' Count: 5

        Group: Approvals
        Total 'verify' Count: 6
        Total 'verifyAll' Count: 2
        Total 'verify*' Count: 8

        Group: Approvals
        Total 'verify' Count: 6
        Total 'verifyAll' Count: 2
        Total 'verify*' Count: 8

        Group: AssertJ
        Total 'assertThat' Count: 22

        Group: Hamcrest
        Total 'assertThat' Count: 62

        Group: All
        Total 'All' Count: 935

```

How to push the above console output in HTML report at the right place in HTML view.. there are little or no examples for such dashboard extensions.

**Comment by portyu9** [comment]
Hi @nagkumar 

The important distinction here is that this HTML report is generated by **Gradle**, not by JUnit Jupiter itself.

Therefore a JUnit 5 extension cannot add a new column or custom section to Gradle's built-in HTML report. Gradle does not expose a supported HTML report-template extension API for doing that.

You can publish your assertion counts from the JUnit side as structured report entries, for example through `TestReporter` or `ExtensionContext.publishReportEntry(...)`, but whether/how those entries are rendered is up to the reporting tool. That does **not** give you a way to insert a custom "assert count" field into Gradle's standard HTML dashboard.

Since you already have the assertion-count data, the supported approaches are:

1. Publish/store the counts during test execution.
2. Generate a separate custom HTML report from that data, or create a Gradle reporting task/plugin that combines it with the test results.

I would avoid using Gradle internal report classes to modify the existing HTML. The old internal test-reporter replacement hook is deprecated and has no supported replacement.

So this is primarily a **Gradle reporting customization** problem rather than a JUnit extension problem.

Please mark this answer as accepted if it helped, thank you.
